scripting/pvpoptin.sp

```diff
@@ -652,23 +675,43 @@ public Action Command_ForcePvP(int client, int args) {
 			if (matches < 1) {
 				ReplyToTargetError(client, matches);
 			} else {
+				bool wasSuccessful = false;
 				for (int i;i<matches;i++) {
 					int player = target[i];
 					if (!IsClientInGame(player)) continue;
-					if (clientPvPBannedUntil[client] > GetTime()) continue; //is banned
-					if (pvpon) {
+					if (clientPvPBannedUntil[player] > GetTime()) continue; //is banned
+					if (pvplock) {
 						globalPvP[player] |= State_Forced;
-						CPrintToChat(player, "%t","Someone forced your global pvp", client);
+						CPrintToChat(player, "%t","Someone locked on your global pvp", client);

```

Comment:
I basically made !forcepvp have two functionalities, locking people into pvp and toggling their personal setting (basically you can "kick" people out of pvp instead of having to ban and unban them)

Reply:
ah ok, I'm not a fan of changing settings for players, that's why force is an override. Like if we have an event that just requires pvp, we require pvp. If it's not required we let the players decide if they want to participate.
I guess there's also [Fake and Force's `/fakecmd`](https://forums.alliedmods.net/showthread.php?p=1625282) as well as [Change overrides](https://forums.alliedmods.net/showthread.php?t=285381) to mess with players temporarily, but I see your the point.
I have to think about this a bit more about this...

Reply:
I had to sometimes use it in rare cases where a player didn't know how to chat or couldn't see chat lol